code/game/objects/items/changeling_vial.dm

```diff
@@ -7,26 +7,29 @@
 	w_class = 1
 	var/genomes_to_give = 10 //seeing as the new changeling won't have had a whole round to prepare, they get some genomes free
 
-/obj/item/changeling_vial/attack_self(mob/user as mob)
-	if(istype(user, /mob/living/carbon/human))
+/obj/item/changeling_vial/attack_self(mob/user)
+	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
 		if(H.mind)
 			var/datum/mind/M = H.mind
 			if(!ischangeling(H))
 				user.visible_message("<span class='warning'>As [user] crushes the vial, a mass of black goo leaps at \his face!.</span>","<span class='warning'>As you try to remove the cap, you crush the vial in your hand!</span> <span class='danger'>A mass of black goo leaps at you from the vial!</span>")
 				playsound(get_turf(src), "shatter", 20, 1)
-				M.make_new_changeling(0,1)
-				H.mind.changeling.geneticpoints = Clamp(genomes_to_give, 0, 100)
 				qdel(src)
 				H.sleeping += 10
 				sleep(100)
 				to_chat(H, "You feel your consciousness slipping away...")
 				sleep(100)
+				var/datum/role/changeling/C = new(M)
+				if(C)
+					C.geneticpoints+=genomes_to_give
```

Comment:
btw you messed this up :^)
(this is the thing I'm fixing in the rebase)

Reply:
aw bugger